File: Olimpiadi Informatica/completati/farmula1.cpp
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <fstream>
#include <iostream>
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    // uncomment the two following lines if you want to read/write from files
    ifstream cin("input0.txt");
    ofstream stampa("output.txt");

    int T;
    cin >> T;
    for (int test = 1; test <= T; ++test) {
        int N;
        cin >> N;
        int cont = 0;
        
        vector<int> P(N);
        for (int i = 0; i < N; ++i) {
            cin >> P[i];
        }
        string ans = "";
        
        
        // INSERT YOUR CODE HERE
        bool won = false;

        for (int i = 0; i < N; i++) {
            if (P[i] == 1) cont++;
        }

        int num = N/2;
        cout << num;

        if (cont >= num) won = true;

        if (won) cout << "Champion";
        else cout << "Practice harder";
    }

    return 0;
}

File: Olimpiadi Informatica/iniziati/farmula1.py
#!/usr/bin/env python3
# NOTE: it is recommended to use this even if you don't understand the following code.

import sys

# uncomment the two following lines if you want to read/write from files
sys.stdin = open('input0.txt')
sys.stdout = open('output.txt', 'w')

T = int(input().strip())
for test in range(1, T+1):
    input()
    N = int(input().strip())
    
    P = list(map(int, input().strip().split()))
    
    ans = ""
    
    
    # INSERT YOUR CODE HERE
    won = False
    cont = 0

    for i in P:
        if i == 1: cont += 1

    num = round(N / 2)
    if ((N % 2) != 0): num = round(N / 2) + 1

    if cont >= num: won = True

    if won: print("Champion")
    else: print("Practice harder")
    
sys.stdout.close()
